std-bits/tableau/src/main/java/org/enso/tableau/HyperReader.java

```diff
@@ -0,0 +1,209 @@
+import com.tableau.hyperapi.HyperProcess;
+import com.tableau.hyperapi.SchemaName;
+import com.tableau.hyperapi.TableName;
+import com.tableau.hyperapi.Telemetry;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.channels.Channels;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.IntStream;
+import org.enso.table.data.table.Column;
+import org.enso.table.problems.ProblemAggregator;
+
+// ** Class responsible for reading from Tableau Hyper files. */
+public class HyperReader {
+  public static final Path HYPER_PATH = Path.of(getHyperPath());
+  private static HyperProcess process;
+
+  private static final Logger LOGGER = Logger.getLogger("enso-hyper-reader");
+
+  private static String getHyperPath() {
+    if (System.getenv("HYPER_PATH") != null) {
+      return System.getenv("HYPER_PATH");
+    }
+    if (System.getenv("ENSO_DATA_DIRECTORY") != null) {
+      return System.getenv("ENSO_DATA_DIRECTORY") + "/hyper";
+    } else {
+      return switch (OSPlatform.CurrentPlatform) {
+        case WINDOWS -> System.getenv("LocalAppData") + "/enso/hyper";
+        case MAC_ARM64, MAX_X64 -> System.getProperty("user.home")
+            + "/Library/Application Support/org.enso/hyper";
+        case LINUX, OTHER -> System.getProperty("user.home") + "/.local/share/enso/hyper";
+      };
+    }
+  }
+
+  private static HyperProcess getProcess() throws IOException {
+    // Check if the hyper directory exists, if not create it.
+    if (!Files.exists(HYPER_PATH)) {
+      try {
+        Files.createDirectories(HYPER_PATH);
+      } catch (Exception e) {
+        throw new IOException("Failed to create Hyper directory: " + HYPER_PATH, e);
+      }
+    }
+
+    // Check if any files in the hyper directory, otherwise download them.
+    try (var files = Files.list(HYPER_PATH)) {
+      if (files.findAny().isEmpty()) {
+        switch (OSPlatform.CurrentPlatform) {
+          case WINDOWS -> downloadHyper(
+              "https://enso-data-samples.s3.us-west-1.amazonaws.com/tableau/hyperd.exe",
+              "hyperd.exe",
+              false);
+          case MAC_ARM64 -> downloadHyper(
+              "https://enso-data-samples.s3.us-west-1.amazonaws.com/tableau/macos-arm64/hyperd",
+              "hyperd",
+              true);
+          case MAX_X64 -> throw new IOException(
+              "Unsupported platform: Only ARM64 Mac is supported.");
+          case LINUX -> downloadHyper(
+              "https://enso-data-samples.s3.us-west-1.amazonaws.com/tableau/linux/hyperd",
+              "hyperd",
+              true);
+          case OTHER -> throw new IOException(
+              "Unsupported platform: " + OSPlatform.CurrentPlatform);
+        }
+      }
```

Comment:
Why cannot this executable be distributed together with the library?

Reply:
It's massive (>200Mb) so making it fetch it when used.

Reply:
Okay, fine for now but I think for longer term we should prefer to keep this as part of the library and try using our existing infrastructure to fetch the whole library on demand.

Also that's why I'd like to put this into a `cache` directory - if we ever switch to distributing with library, a cache dir will get more likely to be cleaned up, while this may stay on user's system for longer.